### PR TITLE
docs: use valid rule config in the custom rule example

### LINKED ISSUE #4721: Docs: Use valid rule config in the custom rule example
### Suggested Changes

The sample code for a [custom rule](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/development/CUSTOM_RULES.md) contains `meta.docs.category`, which is no longer valid.  It is also missing the required `name` ,`defaultOptions`, and `meta.docs.recommended` parameters. The docs should be updated with a sample that passes typechecking out of the box.

### Affected URL(s)

https://typescript-eslint.io/docs/development/custom-rules

### PR DESCRIPTION
**Pull request by alceil**
## PR Checklist

-   [x] Addresses an existing issue: fixes #4721 
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

### CODE REVIEW

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
A good start, thanks for sending this!

Try copy & pasting the code into the [TypeScript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgUQMoBlgDsYFUbAA2AznAL5wBmUEIcA5AAIwCeYApsQMZTBgwBaToWwwA9AFcCJegG4AsACglXCFmLwe7AIYx2AJQmF2cALwoMo-EWIA6Q8YDCUHTGgAKJXDhZtIE6YAfHAABgAWMDBgxABcYmLsAB5+YMa2qiBiUEbsYgAkCL7+ZCEANEoAlArKivFwACps7DFwDuwAshAAJjkAPABEEmAcUFzaxOz9pXC2s4FKSZCwcKrq8NnGZisuugY57ghe267s7qt6iTAViEfeLjASUFg3it5vcABiElhcBGoAIuwuIRtFBdMA1O4sN12NdDq93m9gJQ4O4xAA9ADa2gEAC8ALpiWx6DRQmG2YBdWxFWFw26I7znJIwWwuJYwA70hnefzEYjaADm7AAkl0WvQhiMxhN6OUEdy3tCus0fOTKXKFd4yFUuW8yLqyBq9dUtUb-DBtC14e8uhAuLEXgyXBl-FhlWKGAB3UFYWW65XcXj8CFYGK67z0L4-P7PZXA0HgtQ+PycODEMIQIxdNMW5ae4AwMJwbTPSXsKACaUmYyRcu2P3y8hGnmcflCh3WxFl0bjFX0VC5+CF4CkGlwfOF4ul4blyu9uA1vRQevNpv01gccXECQC9sxhvvbhhdggS1wTH4o2GpTa6pAA) and you'll see it's still missing two properties:

* `defaultOptions`
* `name`

**Comment by alceil** [comment]
Got  it lemme try @JoshuaKGoldberg

**Comment by alceil** [comment]
I have done the required changes please have a look @JoshuaKGoldberg

File: docs/development/CUSTOM_RULES.md [hunk 1]
```diff
@@ -46,27 +46,30 @@ export const rule = createRule({
   create(context) {
     return {
       FunctionDeclaration(node) {
-        if (/^[a-z]/.test(node.id.name)) {
+   if(node.id!=null){
```
Comment (JoshuaKGoldberg) [inline docs/development/CUSTOM_RULES.md:49]: This code isn't formatted very nicely, and [Prettier](https://prettier.io/) is complaining as part of our build checks. That's why this PR has a big red :x: on its status. https://github.com/typescript-eslint/typescript-eslint/runs/5741983136?check_suite_focus=true

I'd recommend opening the repository in VS Code -either by cloning locally or with the GitHub online editor and using the [Prettier VS Code extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) to format it.

Alternately, you can use `prettier --write` on the command-line.

File: docs/development/CUSTOM_RULES.md [hunk 2]
```diff
@@ -46,27 +46,30 @@ export const rule = createRule({
...
           context.report({
             messageId: 'uppercase',
             node: node.id,
           });
         }
+   }
       },
     };
   },
+  name: 'rule-creator',
```
Comment (JoshuaKGoldberg) [inline docs/development/CUSTOM_RULES.md:60]: This `name` is the name of the rule, which is typically 1-3 words summarizing what the rule enforces.

Maybe... `uppercase-first-declarations`? Something like that?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Closer! 🙂

**Comment by JoshuaKGoldberg** [comment]
Btw, you don't need to @ tag me to get a review - we get notifications for this repo's events. Thanks though!

File: docs/development/CUSTOM_RULES.md (hunk 1 above)
Comment (alceil) [reply docs/development/CUSTOM_RULES.md:49]: Ok got it

File: docs/development/CUSTOM_RULES.md (hunk 2 above)
Comment (alceil) [reply docs/development/CUSTOM_RULES.md:60]: ya sure

**Comment by alceil** [comment]
Sure

**Comment by alceil** [comment]
I have done the requested changes.

**Comment by alceil** [comment]
I have a doubt @JoshuaKGoldberg prettier works only on ts extension files right. It doesn't work on markdown files If I'm right. So I just ran prettier to the code snippet I made changes on. Is it alright?

**Comment by JoshuaKGoldberg** [comment]
> work on markdown files

Prettier does work on Markdown files. It's listed on https://prettier.io and https://prettier.io/docs/en/index.html. Prettier will format code inside Markdown code blocks in the language of the code block.

We can see in the PR checks that there is still a formatting issue with the code:

Screenshot of failing PR check: Code style and lint

If you click the _Details_ link there you'll be taken to a page that shows the error log: https://github.com/typescript-eslint/typescript-eslint/runs/5748785507?check_suite_focus=true

```
$ prettier --list-different "./**/*.{md,mdx,ts,js,tsx,jsx}"
docs/development/CUSTOM_RULES.md
```

CI is running the `prettier` command on all `md`, `mdx`, and a few other file extensions. The only failing file is `docs/development/CUSTOM_RULES.md`.

If this is new and inconvenient for you, let me know and I can run Prettier on it for you 🙂. We just don't by default as a lot of folks use pull requests as an opportunity to learn tooling such as Prettier. 

<details>

<summary>
If you do want to do this yourself (yay!), click this for a more explicit list of steps than <a href="https://github.com/typescript-eslint/typescript-eslint/pull/4760#discussion_r837852966">my comment above</a> to edit in an online editor.
</summary>

1. Visit **https://github.dev/alceil/typescript-eslint/tree/patch-1** 
    * (I found that URL by visiting https://github.com/alceil/typescript-eslint/tree/patch-1 and pressing `.` on my keyboard [[github.dev docs](https://github.com/github/dev)])
2. In the bottom-right corner, it should ask you if you want to install recommended extensions. Press **Install**.
    VS Code popup asking to install recommended extensions
3. If anything gives you a warning about limited functionality, press **Install anyway**.
    VS Code popup asking to install markdownlint despite limited functionality
4. Click the file explorer tab on the left, then find and open `docs/development/CUSTOM_RULES.md`
    File explorer open to docs/development/CUSTOM_RULES.md
    * Alternately, use the file selector (<kbd>Cmd</kbd>+<kbd>P</kbd> on <kbd>Mac</kbd>, or  <kbd>Ctrl</kbd>+<kbd>P</kbd> on Windows) to search for that file and <kbd>Enter</kbd> to open it
5. Save the file (shortcut: <kbd>Cmd</kbd>+<kbd>S</kbd> on <kbd>Mac</kbd>, or  <kbd>Ctrl</kbd>+<kbd>S</kbd> on Windows)
6. Click the source control tab on the left, then in the text input, type any message vaguely like `ran prettier`
    Source control with the message 'ran prettier'
7. Click the check icon (label: `Commit`) to make a commit, then the refresh icon just to its right (label: Refresh) to push that commit up to GitHub
    Source control hovering over the Commit icon

</details>

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
I think you ran Prettier with a different set of settings than this repository uses. 🙂

**Comment by alceil** [comment]
How can I use with settings of this repo😅. Sorry I'm new to this so it will take some time to figure it out @JoshuaKGoldberg

**Comment by alceil** [comment]
I ran the command mentioned in your recent comments but still, It has no changes to the markdown file. I researched on the internet about it and it seems many have been facing this issue. So can you run prettier for me @JoshuaKGoldberg

**Review by JoshuaKGoldberg — APPROVED** [review]

**Pull request merged**